### ISSUE TITLE
Client-side caching with Amazon ElastiCache Redis OSS

### ISSUE THREAD

**Issue by kerstinmaier** [redis/rueidis#612]
Hi,
we've successfully used the rueidis client to connect to our locally run redis instances (using dockerhub/bitnami/redis 7.2.4
) and it works both with and without client caching activated.

Now we've tried to get our client code to run with an Amazon ElastiCache Redis OSS instance (exact version being "ElastiCache (Redis OSS) version 7.1 (enhanced)"). I've found no documentation that this provider doesn't support client-side caching in the AWS documentation but whenever we connect with the following parameters 

clientOption = rueidis.ClientOption{
			Password:     [PW_PLACEHOLDER],
			InitAddress:  [ADDRESS_PLACEHOLDER]},
			SelectDB:     redisConfig.DBIndex,
			TLSConfig:    tlsConfig,
			DisableCache: false,
                         }

we get the following error during initialization of the rueidis client: 

Error during initialization: unknown subcommand 'tracking'. Try CLIENT HELP.: [CLIENT TRACKING ON OPTOUT]
ClientOption.DisableCache must be true for redis not supporting client-side caching or not supporting RESP3"

We've also tried setting "AlwaysRESP2: true" assuming that might be the problem, but it still results in the same error.

With "DisableCache:true" everything generally works, just of course without any client caching. Do you have any experience with this provider or hints what the problem with getting the client side caching to work?

kerstin.maier@mercedes-benz.com, Mercedes-Benz Tech Innovation GmbH
[Provider Information](https://github.com/mercedes-benz/foss/blob/master/PROVIDER_INFORMATION.md)

**Comment by rueian** [comment]
Hi @kerstinmaier,

Would you mind providing more details about your cluster? I just tested a cluster on my side and it did recognize the CLIENT TRACKING command.

<img width="1333" alt="截圖 2024-08-20 16 23 07" src="https://github.com/user-attachments/assets/40ef979a-f2f6-41c8-a071-bb55a92586f6">

**Comment by kerstinmaier** [comment]
Hi, thanks for your answer, actually for our initial test we haven't designed an own redis cluster, but used a Redis OSS cache serverless instance as described in the following guide: https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/GettingStarted.html

We created it and a fitting EKS cluster according to the documentation with inbound and outbound rules to allow the traffic, I think the main peculiarity of our setup probably is is that we are connecting to it from a Kubernetes cluster with BYOA (Bring your own account), we've tested that the TCP connection works and ports are open and writing and reading from the cache works without the Client-side caching.

We did have the thought that maybe there is a problem with the traffic going back from AWS to our kubernetes cluster when the "tracking" command is used, but generally both inbound and outbound traffic rules should be fitting. Maybe one question: when initializing the rueidis client, is the connection and inbound/outbound traffic for the "tracking" command already tested? We're missing a bit more information on what actually is the cause of the error, despite debugging as far into the reuidis client code as possible.
